cmd/istio_ca/main.go

```diff
@@ -82,19 +84,22 @@ func init() {
 	flags.StringVar(&opts.signingKeyFile, "signing-key", "", "Specifies path to the CA signing key file")
 	flags.StringVar(&opts.rootCertFile, "root-cert", "", "Specifies path to the root certificate file")
 
-	flags.StringVar(&opts.namespace, "namespace", "",
-		"Select a namespace for the CA to listen to. If unspecified, Istio CA tries to use the ${"+namespaceKey+"} "+
-			"environment variable. If neither is set, Istio CA listens to all namespaces.")
+	flags.StringVar(&opts.namespace, "namespace", "", "The namesapce that CA is running at.")
+	flags.BoolVar(&opts.targetAllNamespace, "all-namespace", false,
+		"Indicates whether to listen to all namespace or just the namespace CA is running at")
+
 	flags.StringVar(&opts.kubeConfigFile, "kube-config", "",
 		"Specifies path to kubeconfig file. This must be specified when not running inside a Kubernetes pod.")
 
 	flags.BoolVar(&opts.selfSignedCA, "self-signed-ca", false,
 		"Indicates whether to use auto-generated self-signed CA certificate. "+
 			"When set to true, the '--signing-cert' and '--signing-key' options are ignored.")
+
 	flags.StringVar(&opts.selfSignedCAOrg, "self-signed-ca-org", "k8s.cluster.local",
 		fmt.Sprintf("The issuer organization used in self-signed CA certificate (default to %s)",
 			selfSignedCAOrgDefault))
 
+	// TODO(wattli): change it to IntVar to make it configurable in yaml.
```

Comment:
Sorry I don't agree with this todo. Right now you can already specify a string that can be parsed as a duration object (e.g. 5m, 10h)

Reply:
I see, reverted.